back/spec/acceptance/users_spec.rb

```diff
@@ -525,465 +545,578 @@
+        end
       end
-    end
 
-    put 'web_api/v1/users/:id' do
-      with_options scope: 'user' do
-        parameter :first_name, 'User full name'
-        parameter :last_name, 'User full name'
-        parameter :email, 'E-mail address'
-        parameter :password, 'Password'
-        parameter :locale, 'Locale. Should be one of the tenants locales'
-        parameter :avatar, 'Base64 encoded avatar image'
-        parameter :roles, 'Roles array, only allowed when admin'
-        parameter :bio_multiloc, 'A little text, allowing the user to describe herself. Multiloc and non-html'
-        parameter :custom_field_values, 'An object that can only contain keys for custom fields for users'
+      get 'web_api/v1/users/:id' do
+        let(:id) { @user.id }
+
+        example 'Get the authenticated user exposes the email field', document: false do
+          do_request
+          json_response = json_parse response_body
+          expect(json_response.dig(:data, :attributes, :email)).to eq @user.email
+        end
       end
-      ValidationErrorHelper.new.error_fields(self, User)
 
-      let(:id) { @user.id }
-      let(:first_name) { 'Edmond' }
+      get 'web_api/v1/users/by_slug/:slug' do
+        let(:user) { create :user }
+        let(:slug) { user.slug }
 
-      describe do
-        let(:custom_field_values) { { birthyear: 1984 } }
-        let(:project) { create(:continuous_project) }
+        example_request 'Get one user by slug' do
+          expect(status).to eq 200
+          json_response = json_parse response_body
+          expect(json_response.dig(:data, :id)).to eq user.id
+        end
 
-        before do
-          if CitizenLab.ee?
-            old_timers = create(:smart_group, rules: [
-              {
-                ruleType: 'custom_field_number',
-                customFieldId: create(:custom_field_number, title_multiloc: { 'en' => 'Birthyear?' }, key: 'birthyear', code: 'birthyear').id,
-                predicate: 'is_smaller_than_or_equal',
-                value: 1988
-              }
-            ])
+        example '[error] Get an unexisting user by slug', document: false do
+          do_request slug: 'unexisting-user'
+          expect(status).to eq 404
+        end
 
-            project.permissions.find_by(action: 'posting_idea')
-              .update!(permitted_by: 'groups', groups: [old_timers])
-          end
+        example_request 'Get a user by slug does not include user block data' do
+          expect(status).to eq 200
+          json_response = json_parse response_body
+          expect(json_response.dig(:data, :attributes)).not_to have_key(:blocked)
+          expect(json_response.dig(:data, :attributes)).not_to have_key(:block_start_at)
+          expect(json_response.dig(:data, :attributes)).not_to have_key(:block_end_at)
+          expect(json_response.dig(:data, :attributes)).not_to have_key(:block_reason)
```

Comment:
This is a new test (I refactored the file, so may not be obvious which tests are new)